### DISCUSSION TITLE
sep.sum_* correction for intersection of pixels and ellipse/circle

### DISCUSSION THREAD

**Discussion by vitorms99** [sep-developers/sep#124] [Q&A]
Hi,

I'm new using SEP and I would like to know if the sep.sum_* routines includes correction for outer pixels that are not fully included in the circle/ellipse. For example, a pixel that half is inside the circle and half outside. Does the sep.sum_* functions apply any correction for these cases? Or it counts every full pixel that is within *or* crossed by the circle/ellipse?

Thank you in advance,
VMS

**Comment by kbarbary** [comment]
Good question. This isn't really explained in the docs. I think the only mention is in the description of the `subpix` argument:

> Subpixel sampling factor. If 0, exact overlap is calculated. Default is 5.

The `sep.sum_` functions all calculate the fraction of the pixel that overlaps the aperture, and weights the pixel value by that fraction.

By default the fraction of the pixel included in the aperture is calculated _approximately_ by splitting the pixel into a 5 x 5 grid and checking whether each subpixel is in the aperture or not. However, there is also an option to calculate the _exact_ overlap fraction using [a bunch of tedious geometry](https://github.com/kbarbary/sep/blob/main/src/overlap.h).
